source/AI.cpp

```diff
@@ -692,21 +693,24 @@ void AI::Step(Command &activeCommands)
 	bool opportunisticEscorts = !Preferences::Has("Turrets focus fire");
 	bool fightersRetreat = Preferences::Has("Damaged fighters retreat");
 	const int npcMaxMiningTime = GameData::GetGamerules().NPCMaxMiningTime();
-	for(const auto &it : ships)
+
+	for_each(parallel::par, ships.begin(), ships.end(), [&](auto &it)
```

Comment:
Lambda functions are nice, but just throwing this large block of code into a lambda function doesn't seem very intuitive.

Reply:
I suppose that depends on your definition of intuitive... the standard functions are designed to work this way, taking a for/for_each and just running the insides in parallel.

(Actually, this is more efficient if you put more code in the lambda, because the overhead of the extra function call becomes less significant.)

Reply:
Okay, this might just become a standard for parallel programming that I didn't know about yet.